ec/src/models/bw6/mod.rs

```diff
@@ -159,22 +173,34 @@ impl<P: BW6Config> BW6<P> {
         }
     }
 
-    fn exp_by_x(mut f: Fp6<P::Fp6Config>) -> Fp6<P::Fp6Config> {
-        f = f.cyclotomic_exp(P::X);
+    fn exp_by_x_plus_1(f: &Fp6<P::Fp6Config>) -> Fp6<P::Fp6Config> {
+        P::exp_by_x(f) * f
+    }
+
+    fn exp_by_x_minus_1(f: &Fp6<P::Fp6Config>) -> Fp6<P::Fp6Config> {
+        P::exp_by_x(f) * &f.cyclotomic_inverse().unwrap()
+    }
+
+    fn exp_by_x_minus_1_div_3(f: &Fp6<P::Fp6Config>) -> Fp6<P::Fp6Config> {
+        let u = P::Fp::from(P::X);
+        let one = P::Fp::one();
+        let three = P::Fp::from(3u32);
         if P::X_IS_NEGATIVE {
-            f.cyclotomic_inverse_in_place();
+            let exp = (u + one) / three;
```

Comment:
Is there no way to hardcode this? We can avoid a field inversion that way. same for the other branch.

Reply:
The best way to hardcode that would probably be some kind of an addition chain. So far added it as BW6Config::X_MINUS_1_DIV_3